### PR TITLE
Document multipart_suggestion derive on SessionSubdiagnostic

### PR DESCRIPTION
**Pull request by Xiretza**
For rust-lang/rust#100970.

### CODE REVIEW

**Comment by Xiretza** [comment]
Waiting for rust-lang/rust#101294.

**Review by davidtwco — APPROVED** [review]
LGTM, r=me once rust-lang/rust#101294 lands.

**Pull request merged**
